BCS instance UT run failed, need to find the reason

```
run acceptance tests of huaweicloud/resource_huaweicloud_bcs_instance_test.go:
=== RUN   TestAccBCSV2Instance_basic
    resource_huaweicloud_bcs_instance_test.go:26: Step 1/1 error: Error running apply: exit status 1
        
        Error: Error waiting for HuaweiCloud CCE cluster to be Available: Resource not found: [GET https://cce.cn-north-4.myhuaweicloud.com/api/v3/projects/0970dd7a1300f5672ff2c003c60ae115/clusters/394dcc20-a15c-11ed-8d34-0255ac100240], error message: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","code":404,"errorCode":"CCE.01404001","errorMessage":"Resource not found.","error_code":"CCE_CM.0003","error_msg":"Resource not found","message":"Cluster not found","reason":"NotFound"}
        
          with huaweicloud_cce_node.test,
          on terraform_plugin_test.tf line 41, in resource "huaweicloud_cce_node" "test":
          41: resource "huaweicloud_cce_node" "test" {
        
    testing_new.go:84: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: Error deleting HuaweiCloud CCE Cluster: Resource not found: [DELETE https://cce.cn-north-4.myhuaweicloud.com/api/v3/projects/0970dd7a1300f5672ff2c003c60ae115/clusters/394dcc20-a15c-11ed-8d34-0255ac100240?delete_sfs=true], error message: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","code":404,"errorCode":"CCE.01404001","errorMessage":"Resource not found.","error_code":"CCE_CM.0003","error_msg":"Resource not found","message":"Cluster not found","reason":"NotFound"}
        
--- FAIL: TestAccBCSV2Instance_basic (466.16s)
=== RUN   TestAccBCSV2Instance_kafka
--- PASS: TestAccBCSV2Instance_kafka (2886.12s)
FAIL
FAIL	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud	3352.307s
FAIL
```

--- a/docs/resources/bcs_instance.md
+++ b/docs/resources/bcs_instance.md
@@ -322,3 +322,31 @@ This resource provides the following timeouts configuration options:
 
 * `create` - Default is 90 minutes.
 * `delete` - Default is 30 minutes.
+
+## Import
+
+The BCS instance can be imported using `id`, e.g.
+
+```bash
+$ terraform import huaweicloud_bcs_instance.test <id>
+```
+
+Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
+API response, security or some other reason.
+The missing attributes include: `delete_storage`, `eip_enable`, `enterprise_project_id`, `fabric_version`,
+`orderer_node_num`, `org_disk_size`, `password` and `volume_type`.
+It is generally recommended running `terraform plan` after importing a instance.
+You can then decide if changes should be applied to the instance, or the resource definition should be updated to
+align with the instance. Also you can ignore changes as below.
+
+```hcl
+resource "huaweicloud_bcs_instance" "test" {
+    ...
+
+  lifecycle {
+    ignore_changes = [
+      delete_storage, eip_enable, enterprise_project_id, fabric_version, orderer_node_num, org_disk_size, password, volume_type,
+    ]
+  }
+}
+```

--- a/huaweicloud/provider.go
+++ b/huaweicloud/provider.go
@@ -19,6 +19,7 @@ import (
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/apigateway"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/apm"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/as"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/bcs"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/bms"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cbh"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cbr"
@@ -781,7 +782,7 @@ func Provider() *schema.Provider {
 			"huaweicloud_as_planned_task":     as.ResourcePlannedTask(),
 
 			"huaweicloud_bms_instance": bms.ResourceBmsInstance(),
-			"huaweicloud_bcs_instance": resourceBCSInstanceV2(),
+			"huaweicloud_bcs_instance": bcs.ResourceInstance(),
 
 			"huaweicloud_cbr_backup_share_accepter": cbr.ResourceBackupShareAccepter(),
 			"huaweicloud_cbr_backup_share":          cbr.ResourceBackupShare(),

--- a/huaweicloud/services/acceptance/bcs/resource_huaweicloud_bcs_instance_test.go
+++ b/huaweicloud/services/acceptance/bcs/resource_huaweicloud_bcs_instance_test.go
@@ -1,4 +1,4 @@
-package huaweicloud
+package bcs
 
 import (
 	"fmt"
@@ -12,10 +12,10 @@ import (
 	"github.com/chnsz/golangsdk/openstack/bcs/v2/blockchains"
 
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
-	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
 )
 
-func TestAccBCSV2Instance_basic(t *testing.T) {
+func TestAccBCSInstance_basic(t *testing.T) {
 	var instance blockchains.BCSInstance
 	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
 	password := fmt.Sprintf("%s%s%d", acctest.RandString(5), acctest.RandStringFromCharSet(3, "!@$%^-_=+[{}]:,./?"),
@@ -23,20 +23,20 @@ func TestAccBCSV2Instance_basic(t *testing.T) {
 	resourceName := "huaweicloud_bcs_instance.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheckEpsID(t) },
-		Providers:    testAccProviders,
-		CheckDestroy: testAccCheckBCSInstanceV2Destroy(),
+		PreCheck:          func() { acceptance.TestAccPreCheckEpsID(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckBCSInstanceDestroy(),
 		Steps: []resource.TestStep{
 			{
-				Config: testBCSInstanceV2_basic(rName, password),
+				Config: testBCSInstance_basic(rName, password),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckBCSInstanceV2Exists(resourceName, &instance),
+					testAccCheckBCSInstanceExists(resourceName, &instance),
 					resource.TestCheckResourceAttr(resourceName, "name", rName),
 					resource.TestCheckResourceAttr(resourceName, "edition", "4"),
 					resource.TestCheckResourceAttr(resourceName, "consensus", "etcdraft"),
 					resource.TestCheckResourceAttr(resourceName, "fabric_version", "2.0"),
 					resource.TestCheckResourceAttr(resourceName, "blockchain_type", "private"),
-					resource.TestCheckResourceAttr(resourceName, "enterprise_project_id", HW_ENTERPRISE_PROJECT_ID_TEST),
+					resource.TestCheckResourceAttr(resourceName, "enterprise_project_id", acceptance.HW_ENTERPRISE_PROJECT_ID_TEST),
 					resource.TestCheckResourceAttr(resourceName, "volume_type", "nfs"),
 					resource.TestCheckResourceAttr(resourceName, "org_disk_size", "100"),
 					resource.TestCheckResourceAttr(resourceName, "security_mechanism", "ECDSA"),
@@ -46,11 +46,26 @@ func TestAccBCSV2Instance_basic(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "peer_orgs.#", "1"),
 				),
 			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"delete_storage",
+					"eip_enable",
+					"enterprise_project_id",
+					"fabric_version",
+					"orderer_node_num",
+					"org_disk_size",
+					"password",
+					"volume_type",
+				},
+			},
 		},
 	})
 }
 
-func TestAccBCSV2Instance_kafka(t *testing.T) {
+func TestAccBCSInstance_kafka(t *testing.T) {
 	var instance blockchains.BCSInstance
 	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
 	password := fmt.Sprintf("%s%s%d", acctest.RandString(5), acctest.RandStringFromCharSet(3, "!@$%^-_=+[{}]:,./?"),
@@ -58,20 +73,20 @@ func TestAccBCSV2Instance_kafka(t *testing.T) {
 	resourceName := "huaweicloud_bcs_instance.test"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:     func() { testAccPreCheckEpsID(t) },
-		Providers:    testAccProviders,
-		CheckDestroy: testAccCheckBCSInstanceV2Destroy(),
+		PreCheck:          func() { acceptance.TestAccPreCheckEpsID(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckBCSInstanceDestroy(),
 		Steps: []resource.TestStep{
 			{
-				Config: testBCSInstanceV2_kafka(rName, password),
+				Config: testBCSInstance_kafka(rName, password),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckBCSInstanceV2Exists(resourceName, &instance),
+					testAccCheckBCSInstanceExists(resourceName, &instance),
 					resource.TestCheckResourceAttr(resourceName, "name", rName),
 					resource.TestCheckResourceAttr(resourceName, "edition", "4"),
 					resource.TestCheckResourceAttr(resourceName, "consensus", "kafka"),
 					resource.TestCheckResourceAttr(resourceName, "fabric_version", "1.4"),
 					resource.TestCheckResourceAttr(resourceName, "blockchain_type", "private"),
-					resource.TestCheckResourceAttr(resourceName, "enterprise_project_id", HW_ENTERPRISE_PROJECT_ID_TEST),
+					resource.TestCheckResourceAttr(resourceName, "enterprise_project_id", acceptance.HW_ENTERPRISE_PROJECT_ID_TEST),
 					resource.TestCheckResourceAttr(resourceName, "volume_type", "efs"),
 					resource.TestCheckResourceAttr(resourceName, "org_disk_size", "500"),
 					resource.TestCheckResourceAttr(resourceName, "security_mechanism", "ECDSA"),
@@ -93,12 +108,12 @@ func TestAccBCSV2Instance_kafka(t *testing.T) {
 	})
 }
 
-func testAccCheckBCSInstanceV2Destroy() resource.TestCheckFunc {
+func testAccCheckBCSInstanceDestroy() resource.TestCheckFunc {
 	return func(s *terraform.State) error {
-		config := testAccProvider.Meta().(*config.Config)
-		client, err := config.BcsV2Client(HW_REGION_NAME)
+		cfg := acceptance.TestAccProvider.Meta().(*config.Config)
+		client, err := cfg.BcsV2Client(acceptance.HW_REGION_NAME)
 		if err != nil {
-			return fmtp.Errorf("Error creating huaweicloud BCS client: %s", err)
+			return fmt.Errorf("error creating BCS client: %s", err)
 		}
 
 		for _, rs := range s.RootModule().Resources {
@@ -109,43 +124,46 @@ func testAccCheckBCSInstanceV2Destroy() resource.TestCheckFunc {
 			id := rs.Primary.ID
 			instance, err := blockchains.Get(client, id).Extract()
 			if err != nil {
+				if _, ok := err.(golangsdk.ErrDefault400); ok {
+					return nil
+				}
 				if _, ok := err.(golangsdk.ErrDefault401); ok {
 					return nil
 				}
 				return err
 			}
 			if instance.Basic.ID != "" {
-				return fmtp.Errorf("%s (%s) still exists", rs.Type, id)
+				return fmt.Errorf("%s (%s) still exists", rs.Type, id)
 			}
 		}
 		return nil
 	}
 }
 
-func testAccCheckBCSInstanceV2Exists(name string, instance *blockchains.BCSInstance) resource.TestCheckFunc {
+func testAccCheckBCSInstanceExists(name string, instance *blockchains.BCSInstance) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		rs, ok := s.RootModule().Resources[name]
 		if !ok {
-			return fmtp.Errorf("Not found: %s", name)
+			return fmt.Errorf("not found: %s", name)
 		}
 
 		id := rs.Primary.ID
 		if id == "" {
-			return fmtp.Errorf("No ID is set")
+			return fmt.Errorf("no ID is set")
 		}
 
-		config := testAccProvider.Meta().(*config.Config)
-		client, err := config.BcsV2Client(HW_REGION_NAME)
+		cfg := acceptance.TestAccProvider.Meta().(*config.Config)
+		client, err := cfg.BcsV2Client(acceptance.HW_REGION_NAME)
 		if err != nil {
-			return fmtp.Errorf("Error creating huaweicloud BCS client: %s", err)
+			return fmt.Errorf("error creating BCS client: %s", err)
 		}
 
 		found, err := blockchains.Get(client, id).Extract()
 		if err != nil {
-			return fmtp.Errorf("Error checking %s exist, err=%s", name, err)
+			return fmt.Errorf("error checking %s exist, err=%s", name, err)
 		}
 		if found.Basic.ID == "" {
-			return fmtp.Errorf("resource %s does not exist", name)
+			return fmt.Errorf("resource %s does not exist", name)
 		}
 
 		instance = found
@@ -153,7 +171,7 @@ func testAccCheckBCSInstanceV2Exists(name string, instance *blockchains.BCSInsta
 	}
 }
 
-func testBCSInstanceV2_base(rName string) string {
+func testBCSInstance_base(rName string) string {
 	return fmt.Sprintf(`
 data "huaweicloud_availability_zones" "test" {}
 
@@ -214,7 +232,7 @@ resource "huaweicloud_cce_node" "test" {
 `, rName, rName, rName, rName, rName)
 }
 
-func testBCSInstanceV2_basic(rName, password string) string {
+func testBCSInstance_basic(rName, password string) string {
 	return fmt.Sprintf(`
 %s
 
@@ -245,10 +263,10 @@ resource "huaweicloud_bcs_instance" "test" {
     ]
   }
 }
-`, testBCSInstanceV2_base(rName), rName, HW_ENTERPRISE_PROJECT_ID_TEST, password)
+`, testBCSInstance_base(rName), rName, acceptance.HW_ENTERPRISE_PROJECT_ID_TEST, password)
 }
 
-func testBCSInstanceV2_kafka(rName, password string) string {
+func testBCSInstance_kafka(rName, password string) string {
 	return fmt.Sprintf(`
 %s
 
@@ -308,5 +326,5 @@ resource "huaweicloud_bcs_instance" "test" {
     ]
   }
 }
-`, testBCSInstanceV2_base(rName), rName, HW_ENTERPRISE_PROJECT_ID_TEST, password, password)
+`, testBCSInstance_base(rName), rName, acceptance.HW_ENTERPRISE_PROJECT_ID_TEST, password, password)
 }

--- a/huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go
+++ b/huaweicloud/services/bcs/resource_huaweicloud_bcs_instance.go
@@ -1,11 +1,15 @@
-package huaweicloud
+package bcs
 
 import (
 	"bytes"
+	"context"
 	"fmt"
+	"log"
 	"strings"
 	"time"
 
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 
@@ -14,19 +18,21 @@ import (
 	"github.com/chnsz/golangsdk/openstack/cce/v3/clusters"
 	"github.com/chnsz/golangsdk/openstack/dms/v2/kafka/instances"
 
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/dms"
-	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
-	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
 )
 
-func resourceBCSInstanceV2() *schema.Resource {
+func ResourceInstance() *schema.Resource {
 	return &schema.Resource{
-		Create: resourceBCSInstanceV2Create,
-		Read:   resourceBCSInstanceV2Read,
-		Update: resourceBCSInstanceV2Update,
-		Delete: resourceBCSInstanceV2Delete,
+		CreateContext: resourceBCSInstanceCreate,
+		ReadContext:   resourceBCSInstanceRead,
+		UpdateContext: resourceBCSInstanceUpdate,
+		DeleteContext: resourceBCSInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
 
 		Timeouts: &schema.ResourceTimeout{
 			Create: schema.DefaultTimeout(90 * time.Minute),
@@ -351,12 +357,13 @@ func resourceBCSInstanceV2() *schema.Resource {
 	}
 }
 
-func resourceBCSInstanceV2Create(d *schema.ResourceData, meta interface{}) error {
-	var newCluster bool = false
-	config := meta.(*config.Config)
-	client, err := config.BcsV2Client(GetRegion(d, config))
+func resourceBCSInstanceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var newCluster = false
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.BcsV2Client(region)
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud Blockchain client: %s", err)
+		return diag.Errorf("error creating Blockchain client: %s", err)
 	}
 	createOpts := blockchains.CreateOpts{
 		CreateNewCluster:    &newCluster,
@@ -374,7 +381,7 @@ func resourceBCSInstanceV2Create(d *schema.ResourceData, meta interface{}) error
 		OrdererNodeNumber:   d.Get("orderer_node_num").(int),
 		TC3Need:             d.Get("tc3_need").(bool),
 		RestfulAPISupport:   d.Get("restful_api_support").(bool),
-		EnterpriseProjectId: GetEnterpriseProjectID(d, config),
+		EnterpriseProjectId: cfg.GetEnterpriseProjectID(d),
 		PeerOrgs:            resourceOrgPeer(d),
 		Channels:            resourceChannelInfo(d),
 		CouchDBInfo:         resourceCouchDBInfo(d),
@@ -383,16 +390,16 @@ func resourceBCSInstanceV2Create(d *schema.ResourceData, meta interface{}) error
 		Kafka:               resourceKafkaCreateInfo(d),
 	}
 
-	if v, err := resourceClusterInfo(config, d.Get("cce_cluster_id").(string), GetRegion(d, config)); err != nil {
-		return fmtp.Errorf("Get cluster information failed: %s ", err)
-	} else {
-		createOpts.ClusterType = "cce"
-		createOpts.CCEClusterInfo = v
+	v, err := resourceClusterInfo(cfg, d.Get("cce_cluster_id").(string), region)
+	if err != nil {
+		return diag.Errorf("get cluster information failed: %s ", err)
 	}
+	createOpts.ClusterType = "cce"
+	createOpts.CCEClusterInfo = v
 
 	res, err := blockchains.Create(client, createOpts).Extract()
 	if err != nil {
-		return fmtp.Errorf("Error creating Blockchain instance: %s ", err)
+		return diag.Errorf("error creating Blockchain instance: %s ", err)
 	}
 
 	d.SetId(res.ID)
@@ -405,50 +412,48 @@ func resourceBCSInstanceV2Create(d *schema.ResourceData, meta interface{}) error
 		Delay:        150 * time.Second,
 		PollInterval: 15 * time.Second,
 	}
-	_, err = stateConf.WaitForState()
+	_, err = stateConf.WaitForStateContext(ctx)
 	if err != nil {
-		return fmtp.Errorf("Error waiting for blockchain instance (%s) status to Normal: %s ", instanceID, err)
+		return diag.Errorf("error waiting for Blockchain instance (%s) status to normal: %s ", instanceID, err)
 	}
 
-	return resourceBCSInstanceV2Read(d, meta)
+	return resourceBCSInstanceRead(ctx, d, meta)
 }
 
-func resourceBCSInstanceV2Read(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	client, err := config.BcsV2Client(GetRegion(d, config))
+func resourceBCSInstanceRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.BcsV2Client(region)
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud Blockchain client: %s", err)
+		return diag.Errorf("error creating Blockchain client: %s", err)
 	}
 
 	instanceID := d.Id()
 	instance, err := blockchains.Get(client, instanceID).Extract()
 	if err != nil {
-		return fmtp.Errorf("Error getting Blockchain instance (%s): %s", instanceID, err)
+		return common.CheckDeletedDiag(d, err, "error getting Blockchain instance")
 	}
-	logp.Printf("[DEBUG] Retrieved Blockchain instance %s: %#v", instanceID, instance)
+	log.Printf("[DEBUG] retrieved Blockchain instance %s: %#v", instanceID, instance)
 
-	d.Set("name", instance.Basic.Name)
-	d.Set("edition", instance.Basic.VersionType)
-	d.Set("blockchain_type", instance.Basic.ServiceType)
-	d.Set("consensus", instance.Basic.Consensus)
-	d.Set("security_mechanism", instance.Basic.SignAlgorithm)
-	d.Set("database_type", instance.Basic.DatabaseType)
-	d.Set("restful_api_support", instance.Basic.IsSupportRestful)
-	d.Set("status", instance.Basic.Status)
-	d.Set("tc3_need", instance.CouchDB != (blockchains.CouchDB{}))
-	d.Set("cluster_type", instance.Basic.ClusterType)
-	d.Set("cce_cluster_id", instance.Basic.ClusterID)
-	d.Set("version", instance.Basic.Version)
-	d.Set("purchase_type", instance.Basic.PurchaseType)
-	d.Set("cross_region_support", instance.Basic.IsCrossRegion)
-	d.Set("rollback_support", instance.Basic.IsSupportRollback)
-	d.Set("old_service_version", instance.Basic.OldServiceVersion)
-
-	portalAddrs := make([]string, len(instance.Basic.AgentPortalAddress))
-	for i, v := range instance.Basic.AgentPortalAddress {
-		portalAddrs[i] = v
-	}
-	d.Set("agent_portal_address", portalAddrs)
+	mErr := multierror.Append(
+		d.Set("name", instance.Basic.Name),
+		d.Set("edition", instance.Basic.VersionType),
+		d.Set("blockchain_type", instance.Basic.ServiceType),
+		d.Set("consensus", instance.Basic.Consensus),
+		d.Set("security_mechanism", instance.Basic.SignAlgorithm),
+		d.Set("database_type", instance.Basic.DatabaseType),
+		d.Set("restful_api_support", instance.Basic.IsSupportRestful),
+		d.Set("status", instance.Basic.Status),
+		d.Set("tc3_need", instance.CouchDB != (blockchains.CouchDB{})),
+		d.Set("cluster_type", instance.Basic.ClusterType),
+		d.Set("cce_cluster_id", instance.Basic.ClusterID),
+		d.Set("version", instance.Basic.Version),
+		d.Set("purchase_type", instance.Basic.PurchaseType),
+		d.Set("cross_region_support", instance.Basic.IsCrossRegion),
+		d.Set("rollback_support", instance.Basic.IsSupportRollback),
+		d.Set("old_service_version", instance.Basic.OldServiceVersion),
+		d.Set("agent_portal_address", instance.Basic.AgentPortalAddress),
+	)
 
 	channelList := make([]map[string]interface{}, len(instance.Channels))
 	for i, v := range instance.Channels {
@@ -458,7 +463,7 @@ func resourceBCSInstanceV2Read(d *schema.ResourceData, meta interface{}) error {
 		}
 		channelList[i] = channel
 	}
-	d.Set("channels", channelList)
+	mErr = multierror.Append(mErr, d.Set("channels", channelList))
 
 	peerList := make([]map[string]interface{}, len(instance.Peer))
 	for i, org := range instance.Peer {
@@ -479,7 +484,7 @@ func resourceBCSInstanceV2Read(d *schema.ResourceData, meta interface{}) error {
 			"address":       address,
 		}
 	}
-	d.Set("peer_orgs", peerList)
+	mErr = multierror.Append(mErr, d.Set("peer_orgs", peerList))
 
 	if instance.CouchDB != (blockchains.CouchDB{}) {
 		couchDBList := make([]map[string]interface{}, 1)
@@ -488,7 +493,7 @@ func resourceBCSInstanceV2Read(d *schema.ResourceData, meta interface{}) error {
 			"password":  d.Get("couchdb.0.password"),
 		}
 		couchDBList[0] = info
-		d.Set("couchdb", couchDBList)
+		mErr = multierror.Append(mErr, d.Set("couchdb", couchDBList))
 	}
 
 	if instance.Basic.Consensus == "kafka" {
@@ -500,28 +505,29 @@ func resourceBCSInstanceV2Read(d *schema.ResourceData, meta interface{}) error {
 			"availability_zone": d.Get("kafka.0.availability_zone"),
 		}
 		kafkaList[0] = info
-		d.Set("kafka", kafkaList)
+		mErr = multierror.Append(mErr, d.Set("kafka", kafkaList))
 	}
 
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceBCSInstanceUpdate(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	// Since delete_obs and delete_storage involve updates but contains cloud service modifications,
+	// an empty udpate function will be set and only the read method will be called
 	return nil
 }
 
-func resourceBCSInstanceV2Update(d *schema.ResourceData, meta interface{}) error {
-	//Since delete_obs and delete_storage involve updates but contains cloud service modifications,
-	//an empty udpate function will be set and only the read method will be called
-	return nil
-}
-
-func resourceBCSInstanceV2Delete(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	bcsClient, err := config.BcsV2Client(GetRegion(d, config))
+func resourceBCSInstanceDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	bcsClient, err := cfg.BcsV2Client(region)
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud Blockchain client: %s", err)
+		return diag.Errorf("error creating Blockchain client: %s", err)
 	}
 	if d.Get("consensus").(string) == "kafka" {
-		dmsClient, err := config.DmsV2Client(GetRegion(d, config))
+		dmsClient, err := cfg.DmsV2Client(region)
 		if err != nil {
-			return fmtp.Errorf("Error creating HuaweiCloud Kafka client: %s", err)
+			return diag.Errorf("error creating kafka client: %s", err)
 		}
 		kafkaName := d.Get("kafka.0.name").(string)
 		listOpts := instances.ListOpts{
@@ -530,21 +536,21 @@ func resourceBCSInstanceV2Delete(d *schema.ResourceData, meta interface{}) error
 		}
 		pages, err := instances.List(dmsClient, listOpts).AllPages()
 		if err != nil {
-			return fmtp.Errorf("Error getting kafka instance in queue (%s): %s", kafkaName, err)
+			return diag.Errorf("error getting kafka instance in queue (%s): %s", kafkaName, err)
 		}
 		res, err := instances.ExtractInstances(pages)
 		if err != nil {
-			return fmtp.Errorf("Error quering HuaweiCloud kafka instances: %s", err)
+			return diag.Errorf("error quering kafka instances: %s", err)
 		}
 		if len(res.Instances) < 1 {
-			return fmtp.Errorf("Error quering kafka, returned no results")
+			return diag.Errorf("error quering kafka, returned no results")
 		}
 		if len(res.Instances) > 1 {
-			return fmtp.Errorf("Error quering kafka, returned more than one result")
+			return diag.Errorf("error quering kafka, returned more than one result")
 		}
 		kafkaID := res.Instances[0].InstanceID
 		if r := instances.Delete(dmsClient, kafkaID); r.Result.Err != nil {
-			return fmtp.Errorf("Error deleting kafka instance (%s): %s ", kafkaID, r.Result.Err)
+			return diag.Errorf("error deleting kafka instance (%s): %s ", kafkaID, r.Result.Err)
 		}
 
 		stateConf := &resource.StateChangeConf{
@@ -555,8 +561,8 @@ func resourceBCSInstanceV2Delete(d *schema.ResourceData, meta interface{}) error
 			Delay:      10 * time.Second,
 			MinTimeout: 3 * time.Second,
 		}
-		if _, err = stateConf.WaitForState(); err != nil {
-			return fmtp.Errorf("Error waiting for instance (%s) to delete: %s", kafkaID, err)
+		if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+			return diag.Errorf("error waiting for instance (%s) to delete: %s", kafkaID, err)
 		}
 	}
 
@@ -570,7 +576,7 @@ func resourceBCSInstanceV2Delete(d *schema.ResourceData, meta interface{}) error
 	}
 
 	if err := blockchains.Delete(bcsClient, deleteOpts, blockchainID).Extract(); err != nil {
-		return fmtp.Errorf("Error deleting HuaweiCloud Blockchain instance (%s): %s", blockchainID, err)
+		return common.CheckDeletedDiag(d, err, "error deleting Blockchain instance")
 	}
 	stateConf := &resource.StateChangeConf{
 		Pending:      []string{"IsDeleting"},
@@ -580,11 +586,10 @@ func resourceBCSInstanceV2Delete(d *schema.ResourceData, meta interface{}) error
 		Delay:        15 * time.Second,
 		PollInterval: 10 * time.Second,
 	}
-	_, err = stateConf.WaitForState()
+	_, err = stateConf.WaitForStateContext(ctx)
 	if err != nil {
-		return fmtp.Errorf("Error waiting for blockchain instance (%s) status to deleted: %s ", blockchainID, err)
+		return diag.Errorf("error waiting for Blockchain instance (%s) status to deleted: %s ", blockchainID, err)
 	}
-	d.SetId("")
 
 	return nil
 }
@@ -593,6 +598,9 @@ func blockchainStateRefreshFunc(client *golangsdk.ServiceClient, instanceID stri
 	return func() (interface{}, string, error) {
 		instance, err := blockchains.Get(client, instanceID).Extract()
 		if err != nil {
+			if _, ok := err.(golangsdk.ErrDefault400); ok {
+				return instance, "IsDeleted", nil
+			}
 			if _, ok := err.(golangsdk.ErrDefault401); ok {
 				return instance, "IsDeleted", nil
 			}
@@ -609,16 +617,16 @@ func blockchainStateRefreshFunc(client *golangsdk.ServiceClient, instanceID stri
 	}
 }
 
-func resourceClusterInfo(config *config.Config, clusterID, region string) (*blockchains.CCEClusterInfo, error) {
+func resourceClusterInfo(cfg *config.Config, clusterID, region string) (*blockchains.CCEClusterInfo, error) {
 	clusterInfo := new(blockchains.CCEClusterInfo)
 
-	client, err := config.CceV3Client(region)
+	client, err := cfg.CceV3Client(region)
 	if err != nil {
-		return clusterInfo, fmtp.Errorf("Error creating HuaweiCloud CCE client: %s", err)
+		return clusterInfo, fmt.Errorf("error creating CCE client: %s", err)
 	}
 	n, err := clusters.Get(client, clusterID).Extract()
 	if err != nil {
-		return clusterInfo, fmtp.Errorf("Error retrieving HuaweiCloud CCE: %s", err)
+		return clusterInfo, fmt.Errorf("error retrieving CCE: %s", err)
 	}
 	clusterInfo.ID = clusterID
 	clusterInfo.Name = n.Metadata.Name

